extensions/shared/components/upgrade-nudge/index.jsx

```diff
@@ -19,7 +19,26 @@ import './store';
 
 import './style.scss';
 
-export const UpgradeNudge = ( { planName, trackEvent, upgradeUrl } ) => (
+const getTitle = ( customTitle, planName ) => {
```

Comment:
I'm interested in feedback on setting a custom title. I didn't want to mess with the plan name setting since this is constructed separately. Is there a better way to handle a custom title?

Reply:
I am thinking this may cause the translators some challenges to keep grammar structure correct in all languages without knowing all the options that customTitle could be (but I am no translations expert so could be wrong) - but haven't managed to think of an alternative yet - will keep thinking about it.

Reply:
the only other option I could think of was to pass two title alternatives into the nudge component in the title prop, eg.

```javascript
				<UpgradeNudge
					plan="value_bundle"
					blockName="core/video"
					title={ { knownPlan: __(
								'Upgrade to %(planName)s to upload videos.',
								'jetpack'
							),
							unknownPlan: __(
								'Upgrade to a paid plan to upload videos.',
								'jetpack'
							),
						} 
					}
					subtitle={ __(
						'Upload unlimited videos to your website and \
						display them using a fast, unbranded, \
						customizable player.',
						'jetpack'
					) }
				/>
```
The the code in the nudge component is simplified to:

```javascript
const getTitle = ( customTitle, planName ) => {
	if ( customTitle ) {
		return planName ? sprintf( customTitle.knownPlan, { planName } ) : customTitle.unknownPlan;
	}

	return planName
		? sprintf( __( 'Upgrade to %(planName)s to use this block on your site.', 'jetpack' ), {
				planName,
				title,
		  } )
		:  __( 'Upgrade to a paid plan to use this block on your site.', 'jetpack' );
};
```

Reply:
This works great, and I agree makes a lot more sense to read, and for translators. Thanks!